### PR TITLE
Bump to sdl12-compat (i.e. sdl 1.2.52)

### PR DESCRIPTION
**Pull request by traversaro**
Switch to sdl12-compat due to its recent release (see https://github.com/conda-forge/sdl-feedstock/issues/5 and https://github.com/libsdl-org/sdl12-compat/issues/53). As this is changing effectively the implementation and the build system, we need to double check the API/ABI concerns, i.e. from https://conda-forge.org/docs/maintainer/knowledge_base.html#moving-from-an-autotools-build-to-a-cmake-build we can write checklist:

API/ABI Checklist:
* [x] 1. Check that both libraries have the same SONAME on linux
* [x] 2. Check that both libraries have the same install name and have the same compatibility and current versions.
* [x] 3. Check that the file list is the same in both.
* [x] 4. Check that you use the same options as the same autoconf build.
* [x] 5. Check that the symbols exported are the same.
* [x] 6. Check that additional packaging information stays the same, e.g. is the same pkg-config information provided.

Usual Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
>  6. Check that additional packaging information stays the same, e.g. is the same pkg-config information provided.

See https://github.com/libsdl-org/sdl12-compat/issues/162 .

**Comment by traversaro** [comment]
>  3. Check that the file list is the same in both.

Checked. The list is the same, except for:
* Man files. However, most packages in conda-forge do not include man files, so I think it is ok not to include them in sdl.
* libSDL.a . However, static libraries should not packaged according to https://github.com/conda-forge/cfep/blob/main/cfep-18.md

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
>  1. Check that both libraries have the same SONAME on linux

Yes, both `Library soname: [libSDL-1.2.so.0]`

>  5. Check that the symbols exported are the same.

The old one has 203, while the new one has 235, but all the symbols in the old one are present in the new one.

**Comment by traversaro** [comment]
In https://github.com/robotology/robotology-superbuild/issues/815 I tested the new package against a legacy program compiled against the old libsdl and the program is working fine.

File: recipe/bld.bat [hunk 1]
```diff
@@ -9,7 +9,6 @@ cmake ^
     -G "Ninja" ^
     -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
     -DCMAKE_BUILD_TYPE=Release ^
-    -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON ^
```
Comment (Tobias-Fischer) [inline recipe/bld.bat:12]: Is this now being taken care of upstream?

**Comment by Tobias-Fischer** [comment]
LGTM - happy to merge if you are. Thanks a lot!

File: recipe/bld.bat (hunk 1 above)
Comment (traversaro) [reply recipe/bld.bat:12]: Yes, the CMake build system of sdl12-compat has proper support for marking Windows symbols as visible.

**Comment by wolfv** [comment]
Awesome!

**Comment by traversaro** [comment]
I wanted to do

> LGTM - happy to merge if you are. Thanks a lot!

I wanted to double check the second point of https://conda-forge.org/docs/maintainer/knowledge_base.html#moving-from-an-autotools-build-to-a-cmake-build, so I was waiting for access to a mac .

**Comment by traversaro** [comment]
> I wanted to do
> 
> > LGTM - happy to merge if you are. Thanks a lot!
> 
> I wanted to double check the second point of https://conda-forge.org/docs/maintainer/knowledge_base.html#moving-from-an-autotools-build-to-a-cmake-build, so I was waiting for access to a mac .

The results are the following (courtesy of @lrapetti):
~~~
$sh test.sh

=== otool -L ./libSDL-1.2.0-1.2.15.dylib ===

./libSDL-1.2.0-1.2.15.dylib:

    @rpath/libSDL-1.2.0.dylib (compatibility version 12.0.0, current version 12.4.0)

    [..]

=== otool -L ./libSDL-1.2.0-1.2.52.dylib ===

./libSDL-1.2.0-1.2.52.dylib:

    @rpath/libSDL-1.2.0.dylib (compatibility version 1.0.0, current version 12.52.0)
    
    [..]
~~~

The install name is indeed the same, while the current version is correctly different. However the `compatibility version` should be the same I guess, and this is not the case.

**Comment by traversaro** [comment]
Apparently the compatibility version was different between the autotools and Xcode build system, see https://github.com/libsdl-org/sdl12-compat/blob/cc52691e4f32ad737f50a0a241e902a63fd7d15f/CMakeLists.txt#L24 . Anyhow, there is an option that we can easily set.

**Comment by traversaro** [comment]
After setting the variable the compatibility version is indeed the same of the older version.
~~~
2022-03-11T13:07:32.8141750Z $PREFIX/lib/libSDL-1.2.0.dylib:
2022-03-11T13:07:32.8143880Z 	libSDL-1.2.0.dylib (compatibility version 12.0.0, current version 12.52.0)
~~~

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Pull request merged**